### DISCUSSION TITLE
Feature Request: Make it possible to convert bigint to inet

### DISCUSSION THREAD

**Discussion by andrewthad** [duckdb/duckdb#10303] [Ideas]
In postgres, users can do things like this to convert a bigint to an inet:

    select '0.0.0.0'::inet + cast(source_ip as bigint) from ...
    
In duckdb, my data source is parquet, and parquet does not have any convert types (or logical types) that indicate that a signed 32-bit or signed 64-bit type is an ipv4 address. So, all queries in duckdb result in decimal numbers. Even though the necessary schema information is missing, it would be nice if I could cast in the query itself (as the very last step, just to improve the presentation of the results). As far as how this is accomplished, I don't really mind. Copying what postgres does would work fine.

An alternative would be supporting duckdb-specific metadata in the parquet file itself so that users in control of their parquet files could explicitly communicate that they intended for numbers to be treated as `inet` if duckdb was used to load the parquet file. That's probably more difficult and less likely to actually be used by users.

**Comment by maurice1408** [comment]
There is always the python way?

```python
import duckdb
from duckdb.typing import *

import ipaddress

def intasipv4(ip_int: int) -> str:

    try:
       ipv4 = str(ipaddress.ip_address(ip_int))
    except AddressValueError:
       ipv4 = None
    return ipv4

def ipv4asint(ip_str: str) -> int:

    try:
       ipv4 = int(ipaddress.ip_address(ip_str))
    except AddressValueError:
       ipv4 = None
    return ipv4

duckdb.create_function("intasipv4", intasipv4, [int], str)
duckdb.create_function("ipv4asint", ipv4asint, [str], int)

res_str = duckdb.sql("SELECT intasipv4(3232235521)")
res_int = duckdb.sql("SELECT ipv4asint('192.168.0.1')")

print(res_str)
print(res_int)
```

```
┌───────────────────────┐
│ intasipv4(3232235521) │
│        varchar        │
├───────────────────────┤
│ 192.168.0.1           │
└───────────────────────┘

┌──────────────────────────┐
│ ipv4asint('192.168.0.1') │
│          int64           │
├──────────────────────────┤
│               3232235521 │
└──────────────────────────┘
```

**Reply by andrewthad** [reply]
Thanks! All the documentation for UDFs on the [Python Function API](https://duckdb.org/docs/api/python/function) page shows this same strategy of defining the python function and then calling `duckdb.sql()` inside of a python script. Is there a way to define the python functions in their own file but then use the DuckDB CLI for interactive querying? Something like this:

    # This is not a real option to duckdb
    duckdb --load-udfs-from /path/to/my/udf.py

**Reply by maurice1408** [reply]
Hi Andrew - so at the moment I do not believe that Python UDFs can be persisted outside of the Python session and then called by the CLI. Maybe that's the real Feature Request?!
